addons/hw_l10n_eg_eta/controllers/main.py

```diff
@@ -0,0 +1,138 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import base64
+import PyKCS11
+import platform
+import json
+
+from passlib.context import CryptContext
+from PyKCS11 import PyKCS11Error
+
+from odoo import http
+from odoo.tools.config import config
+
+crypt_context = CryptContext(schemes=['pbkdf2_sha512', 'plaintext'],
+                             deprecated=['plaintext'])
+
+
+class EtaUsbController(http.Controller):
+
+    def _is_secret_valid(self, token):
+        stored_hash = config.get('middleware_secret')
+        if not stored_hash:
+            # empty password/hash => authentication forbidden
+            return False
+        return crypt_context.verify(token, stored_hash)
```

Comment:
@d-fence for the secret, do you prefer a different or more generic name? e.g. `proxy_access_token` or something?